### DISCUSSION TITLE
Easy setting of priorities without repetition

### DISCUSSION THREAD

**Discussion by hhoeflin** [tweag/nickel#1966] [General]
I was wondering if there is an easier way to set priorities for multiple record fields. When overriding larger number of parameters, setting a priority separately for each gets very wordy. 

Is there a way to set this globally or for a whole record? Or maybe some type of alternative asymmetric merge?

**Comment by yannham** [comment]
At some point, there were additional metadata called _recursive priorities_, which would propagate down a  record (including nested subrecords) to the leaf. So, for example, `{foo = {bar = 1, baz = 2}, qux = "hello"} | rec default` would evaluate to `{foo = {bar | default = 1, baz | default = 2}, qux | default = "hello"}`. Unfortunately their semantics was much less obvious after  implementing RFC005 (see https://github.com/tweag/nickel/pull/1600). It's not impossible to implement, but to be honest there were kinda lost and not much work has been done since to revive them. Do you think such a feature could solve your use-case?

There's been another request for setting priorities programmatically when importing external data: https://github.com/tweag/nickel/discussions/1934.

Finally, while it's not yet documented somewhere, we recently looked at the idea of having a directed merge, think `&>`, that would not be symmetric and behave more like inheritance, so you could do `{foo = bar + 1, bar = 2} &> {bar = 3}` which would give `{foo = 4, bar = 3}`.

So, to answer your original question, there's no such mechanism at the time being, but there are different possible solutions, that either have been already implemented but need a rework (recursive priorities) or at the design stage (directed merge). I would be eager to know which one you think would be the most useful to you.

**Reply by yannham** [reply]
As an addendum: for recursive priorities, I guess what you want is rather the `rec force`, if the goal is to override a bunch of settings: you could do `let overrides | rec force = {...my potentially nested and priority-less overrides..} in config & overrides`
